ferry/transform/same_courses.py

```diff
@@ -2,21 +2,92 @@
 Find historical offerings of a course.
 """
 
-from typing import Any
+from typing import cast, Any
 
+import logging
 import itertools
 import edlib
-import networkx
+import networkx as nx
 import pandas as pd
 from tqdm import tqdm
-import logging
 
+# TODO: I don't quite like these hard-coded constants. Is there a better way
+# to measure similarity that's more context-aware?
```

Comment:
Embeddings ❤️ 

Only worries I have about those are 1. speed and 2. order of tokens. 
I think there is probably some hybrid edit distance-cosine similarity metric out there that preserves order. Needs more research if you're interested

Reply:
That's true... but we are relatively fast complexity-wise now, so we can afford a bit heavier text comparisons.

Reply:
Something we can look into for sure